src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,140 @@
+                                               ".png"))}]]])
+                   token-requirement-list)]]))
+
+(defn is-community-locked? [community]
+  (let [{gates :gates} community]
+    (when (some? gates)
+      (when (vector? (:join gates))
+        (not (is-token-requirement-met? (:read gates)))))))
+
+(defn is-channel-locked? [channel]
+  (let [{gates :gates} channel]
+    (when (some? gates)
+      (if (vector? (:read gates))
+        (not (is-token-requirement-met? (:read gates)))
+        (when (vector? (:write gates))
+          (not (is-token-requirement-met? (:write gates))))))))
+
+(defn token-gating [props]
+  (let [{:keys [community channel]} props
+        type (if (some? community) :community :channel)
+        inner-props (if (= type :community) community channel)
+        name (:name inner-props)
+        emoji (:emoji inner-props)
+        emoji-background-color (:emoji-background-color inner-props)
+        community-avatar (:community-avatar inner-props)
+        gates (:gates inner-props)
+        lock-status (if
+                     (= type :community)
+                      (is-community-locked? inner-props)
+                      (is-channel-locked? inner-props))]
+    [rn/view {:style (get styles :container)}
+     [rn/view {:style (merge
+                       (get styles :header-container)
+                       (if
+                        (= type :community)
+                         (get styles :header-spacing-community)
+                         (get styles :header-spacing-channel)))}
+
+      [rn/view {:style (get styles :header-avatar)}
+       (if (= type :community)
+         [rn/image {:source community-avatar
+                    :style (get styles :header-community-avatar-image)}]
+
+         [channel-avatar/channel-avatar {:big? true
+                                         :locked? lock-status
+                                         :emoji emoji
+                                         :emoji-background-color emoji-background-color}])]
+
+      [text/text {:weight :semi-bold
+                  :number-of-lines 1
+                  :size  :heading-1
+                  :style (get styles :header-text)} (if (= type :community) name (str "# " name))]
+      [button/button {:type :outline
+                      :size 32
+                      :icon :info
+                      :style (get styles :header-info-button)} "info"]]
+
+     (map-indexed (fn [gate-index gate]
+                    [:<>
+                     (when-not (= gate-index 0)
+                       [rn/view {:style (get styles :divider)}])
+                     [token-requirement-list {:gate gate}]]) gates)
+
+     (when (= type :channel)
+       [:<>
+        [button/button {:type :purple :style (get styles :enter-button) :disabled lock-status} "# Enter channel"]
+        [text/text
+         {:size :paragraph-2 :style (get styles :info-text)}
+         "Entering channel will reveal your public addresses to the node owner"]])]))
```

Comment:
I think these strings should also be i18n labels

Reply:
done!